components/ui/AppMegaDropdownMenu.vue

```diff
@@ -1,38 +1,63 @@
 <template>
   <article class="app-mega-dropdown">
-    <button
-      ref="button"
-      class="app-mega-dropdown__button"
-      :class="`app-mega-dropdown__button_${kind}`"
-      @click="toggleOpen"
+    <label
+      ref="searchFieldWrapper"
+      class="app-mega-dropdown__search-field-wrapper"
+      :class="`app-mega-dropdown__search-field-wrapper_${kind}`"
     >
-      <span>{{ placeholder }}</span>
-      <svg class="app-mega-dropdown__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path d="M16 22L6 12l1.4-1.4 8.6 8.6 8.6-8.6L26 12z" /></svg>
-    </button>
+      <input
+        v-model="textOnTheFilter"
+        type="text"
+        class="app-mega-dropdown__search-field-wrapper__input"
+        :placeholder="`${placeholder}`"
+        @focus="onShowContent"
+      >
+      <svg class="app-mega-dropdown__search-field-wrapper__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path d="M16 22L6 12l1.4-1.4 8.6 8.6 8.6-8.6L26 12z" /></svg>
+    </label>
     <div
       v-if="showContent"
       ref="dropdown"
       class="app-mega-dropdown__content-container"
     >
       <nav class="app-mega-dropdown__content">
-        <div v-for="(column, columnIndex) in content" :key="`${columnIndex}`" class="app-mega-dropdown__content-column">
+        <div v-for="(column, columnIndex) in filteredContent" :key="`${columnIndex}`" class="app-mega-dropdown__content-column">
```

Comment:
This can be simplified:

```suggestion
        <div v-for="(column, columnIndex) in filteredContent" :key="columnIndex" class="app-mega-dropdown__content-column">
```